specification/containerservices/resource-manager/Microsoft.ContainerService/2017-07-01/containerService.json

```diff
@@ -0,0 +1,839 @@
+        "Standard_NC24r",
+        "Standard_NC6",
+        "Standard_NV12",
+        "Standard_NV24",
+        "Standard_NV6"
+      ]
+    },
+    "ContainerServiceCustomProfile": {
+      "properties": {
+        "orchestrator": {
+          "type": "string",
+          "description": "The name of the custom orchestrator to use."
+        }
+      },
+      "description": "Properties to configure a custom container service cluster.",
+      "required": [
+        "orchestrator"
+      ]
+    },
+    "KeyVaultSecretRef": {
+      "properties": {
+        "vaultID": {
+          "type": "string",
+          "description": "Key vault identifier."
+        },
+        "secretName": {
+          "type": "string",
+          "description": "The secret name."
+        },
+        "version": {
+          "type": "string",
+          "description": "The secret version."
+        }
+      },
+      "description": "Reference to a secret stored in Azure Key Vault.",
+      "required": [
+        "vaultID",
+        "secretName"
+      ]
+    },
+    "ContainerServiceServicePrincipalProfile": {
+      "properties": {
+        "clientId": {
+          "type": "string",
+          "description": "The ID for the service principal."
+        },
+        "secret": {
+          "type": "string",
+          "description": "The secret password associated with the service principal in plain text."
+        },
+        "keyVaultSecretRef": {
+          "$ref": "#/definitions/KeyVaultSecretRef",
+          "description": "Reference to a secret stored in Azure Key Vault."
+        }
+      },
+      "description": "Information about a service principal identity for the cluster to use for manipulating Azure APIs. Either secret or keyVaultSecretRef must be specified.",
+      "required": [
+        "clientId"
+      ]
+    },
+    "ContainerServiceOrchestratorProfile": {
+      "properties": {
+        "orchestratorType": {
+          "type": "string",
+          "enum": [
+            "Kubernetes",
+            "Swarm",
+            "DCOS",
+            "DockerCE",
+            "Custom"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerServiceOrchestratorTypes",
+            "modelAsString": false
+          },
+          "description": "The orchestrator to use to manage container service cluster resources. Valid values are Kubernetes, Swarm, DCOS, DockerCE and Custom."
+        },
+        "orchestratorVersion": {
+          "type": "string",
+          "description": "The version of the orchestrator to use. You can specify the major.minor.patch part of the actual version.For example, you can specify version as \"1.6.11\"."
+        }
+      },
+      "description": "Profile for the container service orchestrator.",
+      "required": [
+        "orchestratorType"
+      ]
+    },
+    "ContainerServiceMasterProfile": {
+      "properties": {
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "enum": [
```

Comment:
Making sure you understand the risks here, setting modelAsString to false will cause breaking changes in sdks whenever the enum is modified. Please acknowledge.

Reply:
Acknowledged: this value is unlikely to change in the future, and we want to benefit from the [validation](https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/creating-swagger.md#modelasstring) that happens with `"modelAsString": false`.

Reply:
As due diligence we like to point it out in PRs, reduces future pains. 😄